feat: implement sample_orientation_from_CRC workflow

gcapes opened the pull request:
I'm not sure if the `doc` string in the workflow is still ok, or needs updating?

aplowman commented on matflow/data/scripts/mtex/sample_orientations_CRC.m:
```diff
@@ -0,0 +1,81 @@
+function sample_texture_CRC(inputs_JSON_path, outputs_HDF5_path)
```
This seems to work anyway, but for consistency with the other MTEX scripts, can you rename this function to `sample_orientations_CRC` so it matches the file name?

aplowman reviewed (changes requested):
I forgot, could you also update the README table, to show this has now been implemented? Thanks!

gcapes commented:
I've updated the README, but need to wait until this is merged to get the link to the docs. I've added a placeholder for now so it doesn't get forgotten.

aplowman commented:
> I've updated the README, but need to wait until this is merged to get the link to the docs. I've added a placeholder for now so it doesn't get forgotten.

You can predict what the URL will be based on the task schema. It's (objective + method + implementation) joined with hyphens, all lowercase. So in this case it will be:

`https://docs.matflow.io/stable/reference/template_components/task_schemas.html#sample-orientations-from-crc-file-mtex`

It's not the end of the world if we get the prediction wrong, but it's easier to do it here so we don't forget about it. You can also build the docs locally to check what it is.
